Travel Scheduler/View Controllers/MapViewController.m

```diff
@@ -0,0 +1,254 @@
+//
+
+#import "MapViewController.h"
+#import "Place.h"
+#import <CoreLocation/CoreLocation.h>
+#import "TravelSchedulerHelper.h"
+
+@interface MapViewController () <CLLocationManagerDelegate, MKMapViewDelegate>
+
+- (NSArray*)calculateRoutesFrom:(CLLocationCoordinate2D)from to:(CLLocationCoordinate2D)to;
+
+@end
+
+@implementation MapViewController
+
+#pragma mark - View Controller LifeCycle
+
+- (void)viewDidLoad
+{
+    [super viewDidLoad];
+    self.title = @"Trip Overview";
+    [self.navigationController.navigationBar setTitleTextAttributes:
+     @{NSForegroundColorAttributeName:[UIColor darkGrayColor],
+       NSFontAttributeName:[UIFont fontWithName:@"Gotham-Light" size:21]}];
+    [self loadMapView];
+    self.arrayOfAnnotations = [[NSMutableArray alloc] init];
+    self.arrayWithoutDuplicates = [[NSMutableArray alloc] init];
+    [self createHomeAnnotation];
+    int count = 0;
+    for(Place *place in self.placesFromSchedule){
+        if (! [place.name isEqualToString:self.homeFromSchedule.name]){
+            [self.arrayWithoutDuplicates addObject:place];
+        }
+    }
+    [self createPathFromHomeToPlace];
+    for (int k = 0; k < [self.arrayWithoutDuplicates count]; k++){
+        Place *place = self.arrayWithoutDuplicates[k];
+        [self addingPlaceMarkers:place withColor:[UIColor redColor] withInt:k];
+    }
+    [self createPathFromHomeToPlace];
+    self.view = _mainMapView;
+    [_mainMapView setDelegate:self];
+    self.buttonToNavigation = makeScheduleButton(@"Click For Trip Navigation");
+    self.buttonToNavigation.alpha = 1.0;
+    [self.buttonToNavigation addTarget:self action:@selector(navigation) forControlEvents:UIControlEventTouchUpInside];
+    [self.view addSubview:self.buttonToNavigation];
+    [self showRouteFrom:self.arrayOfAnnotations[0] to:self.arrayOfAnnotations[1]];
+}
+
+#pragma  mark - Creating a Home Annotation
+
+- (void)createHomeAnnotation{
+    CLLocationCoordinate2D position = CLLocationCoordinate2DMake([self.homeFromSchedule.coordinates[@"lat"] floatValue], [self.homeFromSchedule.coordinates[@"lng"] floatValue]);
+    MKPointAnnotation *marker = [[MKPointAnnotation alloc] init];
+    [marker setCoordinate:position];
+    [marker setTitle: [NSString stringWithFormat:@"%i. %@", 1,self.homeFromSchedule.name]];
+    [_mainMapView addAnnotation:marker];
+    [self.arrayOfAnnotations addObject:marker];
+}
+
+#pragma mark - Creating start and end paths
+
+- (void) createPathFromHomeToPlace
+{
+    Place *pos1 = self.homeFromSchedule;
+    Place *pos2 = self.arrayWithoutDuplicates[0];
+    [self gettingRouteFromApple:pos1 andSeconPlace:pos2];
+}
+
+- (void) createPathFromLastPlaceToHome
+{
+    Place *pos1 = self.arrayWithoutDuplicates[[_arrayWithoutDuplicates count] - 1];
+    Place *pos2 = self.homeFromSchedule;
+    [self gettingRouteFromApple:pos1 andSeconPlace:pos2];
+    
+}
+
+#pragma mark - View contoller Subviews
+
+- (void)viewWillLayoutSubviews
```

Comment:
Just out of curiosity, why do you use viewWillLayoutSubviews and not layoutSubviews?